api/communication-api/src/main/java/com/cognifide/aet/communication/api/metadata/Operation.java

```diff
@@ -53,7 +53,7 @@ public void addParameters(Map<String, String> parameters) {
   }
 
   public Map<String, String> getParameters() {
-    return parameters != null ? ImmutableMap.copyOf(parameters) : Collections.<String, String>emptyMap();
```

Comment:
Please make sure, that this change does not cause `NPE` when `Operation` instance is created using reflections (e.g. by GSON). Case here is when e.g. processing suite first time and there are no patterns in DB yet. Please check this case.